### PR TITLE
EZP-27443: Display main content tab

### PR DESCRIPTION
**Pull request by maxbaldanza**
https://jira.ez.no/browse/EZP-27443

screen shot 2017-06-23 at 15 18 00

### CODE REVIEW

File: src/bundle/Resources/views/content/tabs/content.html.twig [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{% set languageCode = content.versionInfo.contentInfo.mainLanguageCode %}
+
+{% for contentTypeGroup in contentType.contentTypeGroups %}
```
Comment (bdunogier) [inline src/bundle/Resources/views/content/tabs/content.html.twig]: Not the right element :-)

This is the group(s) the Content Type is part of. You want to iterate on the FieldDefinitions groups, that can be accessed with `$fieldDefinition->fieldGroup`.

You can use `@ezpublish.fields_groups.list` to get the human readable strings (uses translations).

**Review by bdunogier — CHANGES_REQUESTED** [review]

**Comment by bdunogier** [comment]
Unfortunately, large changes are required, since the wrong domain objects were used.

File: src/bundle/Resources/views/content/tabs/content.html.twig (hunk 1 above)
Comment (maxbaldanza) [reply src/bundle/Resources/views/content/tabs/content.html.twig]: 🤔 if you look at `UiContentType:: getFieldDefinitionsByFieldGroup` I am using the `$fieldDefinition->fieldGroup` I thought the translated string could be obtained from the content type but I shall take a look at the mentioned service 👍

File: src/bundle/Resources/views/content/tabs/content.html.twig [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    {% set contentTypeName = contentTypeGroup.name(languageCode) %}
+
+    <section class="ez-fieldgroups ez-fieldgroup ez-fieldgroup">
```
Comment (dpobel) [inline src/bundle/Resources/views/content/tabs/content.html.twig]: too many classes from v1 here :)
For now, the following markup should be enough:
```html
<section class="ez-fieldgroup">
    <h3 class="ez-fieldgroup-name">{{ fieldgroupName }}</h3>
    <!-- fields here -->
</section>
```

File: src/lib/Repository/UiFieldGroupService.php [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+     * @return UiFieldGroup[]
+     */
+    public function loadFieldGroups(ContentInfo $contentInfo)
+    {
+        $contentType = $this->contentTypeService->loadContentType($contentInfo->contentTypeId);
+        $fields = $contentType->getFieldDefinitions();
+
+        $fieldGroups = [];
+        foreach ($this->fieldsGroupsList->getGroups() as $groupId => $groupName) {
+            $fieldsInGroup = array_filter($fields, function (FieldDefinition $field) use ($groupId) {
```
Comment (bdunogier) [inline src/lib/Repository/UiFieldGroupService.php]: Would you mind refactoring this part to a private method ? It has quite a lot of logic and not much explanations. Something like `$fieldsInGroup = filterFieldsFromGroup($fields, $groupId);`.

**Review by bdunogier — CHANGES_REQUESTED** [review]

**Review by dpobel — APPROVED** [review]

**Review by bdunogier — APPROVED** [review]

**Pull request merged**